app/src/main/java/com/duckduckgo/app/trackerdetection/TrackerDetector.kt

```diff
@@ -32,12 +31,17 @@ class TrackerDetector @Inject constructor(private val networkTrackers: TrackerNe
 
     private val clients = CopyOnWriteArrayList<Client>()
 
+    /**
+     * Adds a new client. If the client's name matches an existing client, old client is replaced
+     */
     fun addClient(client: Client) {
+        val existingClient = clients.firstOrNull { client.name == it.name }
+        if (existingClient != null) {
+            removeClient(existingClient)
+        }
```

Comment:
You can replace these 3 lines with:

`clients.removeAll { client.name == it.name }`

Reply:
... or do that in the `removeClient` func and just call removeClient every time

Reply:
and I just noticed that `TrackerDataLoader.kt` still calls removeClient for the disconnect list. I would be tempted to remove the removeClient func all together, but we talked about doing a bit of a refactor later so happy to leave it.

Reply:
Good idea. I've killed the removeClient method, we really don't need it.